### ISSUE TITLE
About Loss and function: compute_multiple_prototypes

### ISSUE THREAD

**Issue by lxr-1204** [YazhouZhu19/RPT#19]
I recently had the pleasure of reading your paper and going through your code. It’s an excellent piece of work. I have two questions I’d like to ask:

1. Regarding the loss function, we noticed that there are significant differences between the loss function in your code and the one mentioned in the paper. However, I’m particularly interested in understanding the fundamental differences between the `query_loss` and the `qry_loss` in [train.py](https://github.com/YazhouZhu19/RPT/blob/13b4a4b27ff5624e6b3fe4abf331b3e6d9a785c7/train.py#L121) as they both seem to be cross-entropy losses for the query.

2. In the `compute_multiple_prototypes` [method](https://github.com/YazhouZhu19/RPT/blob/13b4a4b27ff5624e6b3fe4abf331b3e6d9a785c7/models/fewshot.py#L270), when handling the case where `fg_mask_i.sum() < fg_num`, according to the code `fg_mask_i.view(-1)[:fg_num] = True`, it appears that you randomly selected some pixels and set them as foreground. This approach seems like it could introduce a considerable amount of error. What’s your perspective on this?

Thank you for your time and looking forward to your response.
